### DISCUSSION TITLE
HTTPSConnectionPool(host='api.deepgram.com', port=443)

### DISCUSSION THREAD

**Discussion by okyznetsov** [deepgram/community#635] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Congratulations. 

I need some help. I am using Python to get the value of a wav file. If I use the "language": "uk" option, everything works fine. But if I start using the parameters "smart_format": True, "punctuate": True, "paragraphs": True, then I get an error: 

```bash HTTPSConnectionPool(host='api.deepgram.com', port=443): Max retries exceeded with url: /v1/listen?punctuate=True (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2384)'))).```

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen

### If possible, please attach your code or paste it into the text box.

```python
import os
import requests
from dotenv import load_dotenv

load_dotenv()

AUDIO_FILE = "00124_output.wav"
DG_API_KEY = os.getenv('DG-API-KEY')

def main():
    try:
        headers = {
            "Authorization": f"Token {DG_API_KEY}",
            "Content-Type": "audio/wav"
        }

        options = {    
            "smart_format": True,
            "punctuate": True,
            "paragraphs": True,
            "language": "uk",
            "model": "nova-2"
        }

        with open(AUDIO_FILE, 'rb') as file:
            data = file.read()

        response = requests.post("https://api.deepgram.com/v1/listen", data=data, headers=headers, params=options)
        response.raise_for_status()

        transcript = response.json()["results"]["channels"][0]["alternatives"][0]["transcript"]
        print("Транскрипція аудіофайлу:")
        print(transcript)
    except requests.exceptions.RequestException as e:
        print("Сталася помилка під час виконання запиту:")
        print(e)

if __name__ == '__main__':
    main()
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
> But if I start using the parameters "smart_format": True, "punctuate": True, "paragraphs": True, then I get an error:

[Smart Forma](https://developers.deepgram.com/docs/smart-format)t does punctuation and paragraphs for you so you shouldn't need to use both the `smart_format` and `paragraph` and `punctuation` options in this request.

 Is there a reason you are needing apply these parameters twice?
